File: oeml-sdk/dart-jaguar.yaml/lib/model/message_reject.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


import 'package:openapi/model/reject_reason.dart';

part 'message_reject.jser.dart';

class MessageReject {
   /* Message type, constant. */
  @Alias('type', isNullable: false,  )
  final String type;
  
  @Alias('reject_reason', isNullable: false,
          
             processor:  const RejectReasonFieldProcessor(),
          
  )
  final RejectReason rejectReason;
  //enum rejectReasonEnum {  OTHER,  EXCHANGE_UNREACHABLE,  EXCHANGE_RESPONSE_TIMEOUT,  ORDER_ID_NOT_FOUND,  INVALID_TYPE,  METHOD_NOT_SUPPORTED,  JSON_ERROR,  }; /* If the message related to exchange, then the identifier of the exchange will be provided. */
  @Alias('exchange_id', isNullable: false,  )
  final String exchangeId;
   /* Message text. */
  @Alias('message', isNullable: false,  )
  final String message;
   /* Value of rejected request, if available. */
  @Alias('rejected_message', isNullable: false,  )
  final String rejectedMessage;
  

  MessageReject(
      

{
     this.type = null,  
     this.rejectReason = null,  
     this.exchangeId = null,  
     this.message = null,  
     this.rejectedMessage = null 
    
    }
  );

  @override
  String toString() {
    return 'MessageReject[type=$type, rejectReason=$rejectReason, exchangeId=$exchangeId, message=$message, rejectedMessage=$rejectedMessage, ]';
  }
}

@GenSerializer(nullableFields: true)
class MessageRejectSerializer extends Serializer<MessageReject> with _$MessageRejectSerializer {

}

File: ems-cloud-mgmt-sdk/javascript-closure-angular/API/Client/MessageReject.js
goog.provide('API.Client.MessageReject');

/**
 * @record
 */
API.Client.MessageReject = function() {}

/**
 * Message type, constant.
 * @type {!string}
 * @export
 */
API.Client.MessageReject.prototype.type;

/**
 * @type {!API.Client.RejectReason}
 * @export
 */
API.Client.MessageReject.prototype.rejectReason;

/**
 * If the message related to exchange, then the identifier of the exchange will be provided.
 * @type {!string}
 * @export
 */
API.Client.MessageReject.prototype.exchangeId;

/**
 * Message text.
 * @type {!string}
 * @export
 */
API.Client.MessageReject.prototype.message;

/**
 * Value of rejected request, if available.
 * @type {!string}
 * @export
 */
API.Client.MessageReject.prototype.rejectedMessage;

